packages/p2p/src/utils/orders.js

```diff
@@ -1,177 +1,273 @@
-            offered_amount: order_details.amount_display,
-            offered_currency: order_details.account_currency,
+class ExtendedOrderDetails {
+    constructor(order_details, loginid, server_time) {
+        this.order_details = order_details;
+        this.loginid = loginid;
+        this.server_time = server_time;
+
+        // Assign all original props to this.
+        Object.keys(this.order_details).forEach(key => {
+            this[key] = order_details[key];
         });
     }
 
-    return labels;
-};
+    // Order statuses
+    get is_buy_order() {
+        return this.order_details.type === 'buy';
+    }
+    get is_sell_order() {
+        return this.order_details.type === 'sell';
+    }
+    get is_buyer_cancelled_order() {
+        return this.order_details.status === 'cancelled';
+    }
+    get is_buyer_confirmed_order() {
+        return this.order_details.status === 'buyer-confirmed';
+    }
+    get is_completed_order() {
+        return this.order_details.status === 'completed';
+    }
+    get is_expired_order() {
+        return this.order_details.status === 'timed-out';
+    }
+
+    get is_pending_order() {
+        return this.order_details.status === 'pending';
+    }
 
-export const getExtendedOrderDetails = (order_details, loginid) => {
-    if (!loginid) {
-        throw new Error('Missing required loginid');
+    get is_refunded_order() {
+        return this.order_details.status === 'refunded';
     }
 
-    const { advert_details, advertiser_details, client_details, status, type } = order_details;
+    get is_my_ad() {
+        return this.order_details.advertiser_details.loginid === this.loginid;
+    }
 
-    const is_buy_order = type === 'buy';
-    const is_sell_order = type === 'sell';
+    get is_inactive_order() {
+        return this.is_buyer_cancelled_order || this.is_refunded_order || this.is_completed_order;
+    }
 
-    const is_buyer_cancelled_order = status === 'cancelled';
-    const is_buyer_confirmed_order = status === 'buyer-confirmed';
-    const is_completed_order = status === 'completed';
-    const is_expired_order = status === 'timed-out';
-    const is_pending_order = status === 'pending';
-    const is_refunded_order = status === 'refunded';
+    get is_active_order() {
+        return !this.is_inactive_order;
+    }
 
-    const is_my_ad = loginid === advertiser_details.loginid;
+    get is_finalised_order() {
+        return this.is_completed_order || this.is_buyer_cancelled_order || this.is_refunded_order;
+    }
 
-    const is_active_order = is_pending_order || is_buyer_confirmed_order || is_expired_order;
-    const is_inactive_order = is_buyer_cancelled_order || is_refunded_order || is_completed_order;
+    // A happening order describes an order where an action has been taken by either side, i.e.
+    // one side confirmed they've paid or received funds.
+    get is_happening_order() {
```

Comment:
This is just a suggestion, `is_ongoing_order`?

Reply:
Hehe I was struggling with a good name. Ongoing looks much better 👍